jumpscale/sals/backupjob/backupjob.py

```diff
@@ -0,0 +1,223 @@
+"""
+----------------------------------------------------------------------
+# THE BACKUPJOB SAL
+----------------------------------------------------------------------
+This sal can be used to create and manage multiple backup jobs with multiple and configured paths.
+
+Examples:
+1. create new backup job
+    - every package could create its backup job when installed with one or multiple paths
+
+```python
+JS-NG> nginxbackup = j.sals.backupjob.new("nginxbackup", clients = ["restic_client_1", "restic_client_2"], paths=["~/sandbox/cfg/nginx/main/"])
+JS-NG> nginxbackup.save()
+```
+
+2. create another backup job
+JS-NG> vdcbackup = j.sals.backupjob.new("vdcbackup", clients = ["restic_client_3", "restic_client_4"], paths=["~/.config/jumpscale/secureconfig/jumpscale/sals/vdc/"])
+JS-NG> vdcbackup.save()
+
+3. list backup jobs
+
+```python
+JS-NG> j.sals.backupjob.list_all()
+```
+
+4. get and execute a backup job
+    - get the backup job
+    ```python
+    JS-NG> nginxbackup_job = j.sals.backupjob.get('nginxbackup')
+    JS-NG> nginxbackup_job.execute()
+    ```
+"""
+from gevent.greenlet import Greenlet
+from jumpscale.loader import j
+from jumpscale.core.base import Base, fields
+from jumpscale.sals import fs
+import gevent
+
+
+def _path_validator(path):
+    """Raises a ValidationError if a path Neither an absolute path nor begin with a tilde.
+
+    Args:
+        path (str): a path represent a directory/file.
+
+    Raises:
+        j.core.base.fields.ValidationError: If a path Neither an absolute path nor begin with a tilde.
+    """
+    if not fs.is_absolute(fs.os.path.expanduser(path)):
+        raise j.core.base.fields.ValidationError(f"The path {path} should be absolute path or begin with a tilde")
```

Comment:
we can replace with `j.exceptions.Validation(`

Reply:
https://github.com/threefoldtech/js-ng/blob/fde0b260508da92c573dcefd5067e4b3cbf1b518/jumpscale/core/base/fields.py#L95

based in the implementaion, Field validator should raise j.core.base.fields.ValidationError

Reply:
no problem, both works it's just an empty interface, i suggested that instead of long line